### ISSUE TITLE
[IDEA] Extensible subfilter for Recent SideBar timeline

### ISSUE THREAD

**Issue by saqimtiaz** [Jermolene/TiddlyWiki5#6610]
The `timeline` macro used in the sidebar supports a `subfilter` parameter however this is parameter is not used in the core sidebar tab Recent. Note that in this macro the `subfilter` parameter accepts filter **step(s)** and not a complete filter run or expression.

It is proposed to add a `subfilter` parameter to the timelime macro invocation in the Recent tab as follows, so that we feed it a reduction of tagged tiddlers that provide filter step(s):

```
<$macrocall 
	$name="timeline" 
	format={{$:/language/RecentChanges/DateFormat}}
	subfilter={{{ [tag[$:/tags/SideBar/Recent/filtersteps]get[text]join[]] }}}
/>
```

Example user tiddler providing a filter step that excludes all tiddlers starting with the prefix `_`:
```
title: $:/config/SideBar/Recent/exclude
tags: $:/tags/SideBar/Recent/filtersteps

!prefix[_]
```

This will allow users and plugins to exclude tiddlers from the Recent tab without overwriting any shadow tiddlers.

**Comment by saqimtiaz** [comment]
If we don't want to introduce another system tag convention, a smaller but still helpful step would be to set the `subfilter` parameter for the `timeline` macro in `$:/core/ui/SideBar/Recent` to a hidden settings tiddler such as `$:/config/SideBar/Recent/subfilter`. This will allow users to customize the subfilter without overriding the shadow tiddler. However, it wont help with plugins being able to customize the subfilter without stepping on each other.

**Comment by pmario** [comment]
TLDR;

My question is, would it be better to create a new "timelineXX" macro that implements a "real" subfilter, that can be "stacked"

-----

IMO we introduced the cascade system lately, to be more flexible for pluing authors. .. We already have a bunch of system tags. Users know them and I think they understand the mechanism well. 

So I think 1 more doesn't hurt, if we document it. ... 

The only problem I see, is the functionality / implementation itself.

>Note that in this macro the subfilter parameter accepts filter step(s) and not a complete filter run or expression.

I did play with the recent tab some time ago and none of my subfilters worked the way I wanted it, _until_ I did have a closer look at the macro code itself. ... I did have a look at the docs, but that doesn't help until you see, how the "subfilter snippet" is included.

```
<$list filter="[sameday:$dateField${!!$dateField$}!is[system]$subfilter$!sort[$dateField$]]">
```

**Comment by saqimtiaz** [comment]
> I did play with the recent tab some time ago and none of my subfilters worked the way I wanted it, _until_ I did have a closer look at the macro code itself. ... I did have a look at the docs, but that doesn't help until you see, how the "subfilter snippet" is included.

Right, that is a documentation problem more so than an implementation one as the term subfilter is not appropriate at all. That needs to be fixed regardless of any other changes.

Using filter steps isn't necessarily a bad idea here as you only want to allow selection modifiers and not constructors.
